scripts/miniapps/run_miniapps

```diff
@@ -0,0 +1,264 @@
+    launch_args_snap_large="-n 8192 -N 128 -c 2 --threads-per-core=2 --exclusive"
+    launch_args_lulesh_small="-n 8 -N 2 --ntasks-per-node=4 -c 2 --threads-per-core=2 --exclusive"
+    launch_args_lulesh_medium="-n 216 -c 2 --threads-per-core=2 --exclusive"
+    timeout_1="-t00:15:00"
+    timeout_2="-t00:20:00"
+    timeout_3="-t00:25:00"
+else
+    launch_args_small="-n 16 -N 8 -d 2 -cc depth"
+    launch_args_medium="-n 128 -d 2 -cc depth"
+    launch_args_large="-n 8000 -N 64 -d 2 -cc depth -j 2"
+    launch_args_snap_small="-n 6 -N 3 -d 2 -cc depth"
+    launch_args_snap_medium="-n 96 -N 24"
+    launch_args_snap_large="-n 8192 -N 64 -d 2 -cc depth -j 2"
+    launch_args_lulesh_small="-n 8 -N 4 -d 2 -cc depth"
+    launch_args_lulesh_medium="-n 216 -d 2 -cc depth -j 2"
+    timeout_1="-t 900"
+    timeout_2="-t 1200"
+    timeout_3="-t 1500"
+fi
+
+ret=0
+not_run=0
+pass=0
+fail=0
+
+run_test() {
+    test=$1
+    l_args="$2"
+    limit="$3"
+    app_args="$4"
+
+    if [ ! -x $test ]; then
+	blah=$((not_run++))
+	echo "$test not found."
+	ret=-1
+	continue
+    fi
+    echo "###"
+    echo "### ldd ./$test" 
+    echo "###"
+    ldd ./$test
+    echo ""
+    echo "###"
+    echo "### Running: $launcher $l_args $limit ./$test $app_args"
+    echo "###"
+    $launcher $l_args $limit ./$test $app_args
+    if [ $? != 0 ]; then
+	echo "$test FAILED ($l_args $limit ./$test $app_args)"
+	blah=$((fail++))
+	ret=-1
+    else
+	blah=$((pass++))
+	echo ""
+    fi
+}
+
+#
+# DOE Mini Apps XSBench, SNAP, HPGMG, LULESH, CoMD
+#
+
+# For this group of apps, select any or all of these
+PowerOfTwo="
+XSBench
+hpgmg-fv
+"
+
+date
+echo "###"
+echo "### Mini Apps which no special nranks requirements (XSBench, HPGMG)"
+echo "###"
+
+for test in $PowerOfTwo; do
+    if [ $size = "large" ]; then
+        launch_args="$launch_args_large"
+        timeout="$timeout_2"
+    else if [ $size = "medium" ]; then
+        launch_args="$launch_args_medium"
+        timeout="$timeout_1"
+    else
+        launch_args="$launch_args_small"
+        timeout="$timeout_1"
+    fi
+    fi
+    export OMP_NUM_THREADS='2'
+    #echo $OMP_NUM_THREADS
+    if [ $test = "XSBench" ]; then
+        app_args="-s small -t 2 -l 1000"
+    else if [ $test = "hpgmg-fv" ]; then
+        app_args="6 8"
+    fi
+    fi
+    run_test $test "$launch_args" "$timeout" "$app_args"
+done
+
+# For the next group of apps, select any or all of these
+Special="
+snap
+CoMD-openmp-mpi
+lulesh2.0
+"
+#gsnap
+#ksnap
+#isnap
+
+echo "###"
+echo "### Mini Apps with strict launch requirements, varying app parameters etc. (SNAP, LULESH, CoMD)"
+echo "###"
+
+for test in $Special; do
+    #if [ $test = "gsnap" ] || [ $test = "ksnap" ] || [ $test = "isnap" ]; then
+    if [ $test = "snap" ] ; then
+        if [ $size = "large" ]; then
+            launch_args="$launch_args_snap_large"
+            app_args="./8192MT.input ./SNAPlarge.output"
```

Comment:
These args assume that the input decks are in `$APP_DIR` and the output files are also written to `$APP_DIR`.  I don't think that's the way it should be.  I think the input decks should be found in the same directory as the script itself (e.g., `SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)`) and that the output files should be in the directory that the script is called from.